app/controllers/admin/admin_controller.rb

```diff
@@ -1,34 +1,33 @@
 module Admin
   class AdminController < ApplicationController
     before_action :authorized?
+    before_action :retrieve_source_model_by_id, only: [:edit, :update, :destroy]
+    before_action :retrieve_source_table, only: [:index, :new, :create]
```

Comment:
you don't need to do this. just have a function

```
def source_klass
  @source_klass ||= ||= controller_name.singularize.classify.constantize
end

```

and then you can call source_klass as many times as you want. 

Its generally a good idea to wrap the access to @vars like this as it yields cleaner/more readable code. IE up here at the top of the file you're using a variable named @source_table but when reading the code that isn't actually defined yet so it causes confusion and the possibility for a developer to attempt to use the variable before its defined.